homeassistant/components/rova/sensor.py

```diff
@@ -27,36 +31,47 @@
 SCAN_INTERVAL = timedelta(hours=12)
 
 
-class RovaSensorMetadata(NamedTuple):
+@dataclass
+class RovaSensorEntityDescription(SensorEntityDescription):
     """Metadata for an individual rova sensor."""
 
-    name: str
-    json_key: str
-    icon: str
+    json_key_required: str | None = None
+
+    def __post_init__(self) -> None:
+        """Check required keys are present."""
+        if self.json_key_required is None:
+            # pragma: no cover
+            raise TypeError
+        self.json_key = self.json_key_required
```

Comment:
My current solution for the required key problem. `self._json_key` will be inferred as just `str`.

Reply:
If we keep the dict per my comment above and use the name attribute of the description below we only need one description key for the json access.

Reply:
Unless I'm missing something, that doesn't really solve the problem as the `name` field is also defined as `str | None` (same with `json_key_required`). The issue at hand is that dataclasses don't allow fields without defaults in derived classes if one of the bases has at least one field with a default value.

Reply:
Is this just a construct to change the typing of the key?

Reply:
It's one idea to do so. A normal cast would work too, but that doesn't check if `json_key_required is not None`. I'm open to other ideas how to resolve it.